### DISCUSSION TITLE
Vue 3 Compatibility

### DISCUSSION THREAD

**Discussion by dhershman1** [dhershman1/vue-debounce#47] [Ideas]
**Compatability added for vue 3 in v2.6.0 of vue-debounce**

So with the release of Vue 3 they're changing the plugin function from `bind` to `mounted`

This obviously means that vue-debounce wouldn't work on vue 3 projects

It's also worth noting that as of this (11/18/2020) Vue 3 is still on the `next` branch on npm and not on the main branch, so time isn't a huge deal here, but I'd like to get this started sooner rather than later.

There are two ways we could go about this:

1. The next major release of vue-debounce (3.0.0) switches to this new plugin function, this means v3.0.0 and on won't work on vue 2 projects anymore. But I think vue-debounce v2+ seems stable enough.

2. I build in some backwards compatibility functionality, basically checking which version of Vue the user is using, and then swap out the plugin function name on install. This works fine and dandy but makes me a little uneasy at the same time. It would be something like

```js
function determineVersion (v) {
  const [major] = v.split('.')

  if (Number(major) >= 3) {
    return 'mounted'
  }

  return 'bind'
}
```

These are my current ideas, feel free to comment about which approach you think would be better, or if you have your own idea I'd also love to hear it so comment it too!

**Comment by dani-fmena** [comment]
guys, which is the ETA for this ?

PS: The 2nd approach is fine IMHO.

**Comment by dhershman1** [comment]
> guys, which is the ETA for this ?
> 
> PS: The 2nd approach is fine IMHO.

As soon as I get some more feedback. Lol I mean unless we don't then I guess your feedback is the only one that matters and we go that route! 😆

**Comment by dhershman1** [comment]
v2.6.0 released with vue 3 compatibility introduced. 

https://github.com/dhershman1/vue-debounce/releases/tag/v2.6.0
